File: src/main/java/my/study/repository/design/pattern/flyweight/UseDemo.java
package my.study.repository.design.pattern.flyweight;

/**
 * 享元模式: 关键字 "享".
 * <p>
 * This patter is use to when the system have a lot of repeat object, so we should reuse
 * them not is rebuild, that will let the gc very frequently, and let the efficiency
 * reduce.
 *
 * @author djh on  2018/9/19 22:26
 * @E-Mail 1544579459@qq.com
 */
public class UseDemo {
    public static void main(String[] args) {
        Ticket ticket = TicketFactory.getTicket("北京", "上海");
        ticket.showTicketInfo("上铺");
        Ticket ticket1 = TicketFactory.getTicket("北京", "上海");
        System.out.println(ticket == ticket1);
    }
}

File: src/main/java/my/study/repository/design/pattern/flyweight/kotlin/UseDemo.kt
package my.study.repository.design.pattern.flyweight.kotlin

/**
 * @author djh on  2018/10/5 19:02
 * @E-Mail 1544579459@qq.com
 */
object UseDemo {
    @JvmStatic
    fun main(args: Array<String>) {
        val ticket = getTicket("上海", "北京")
        ticket.showTicketInfo("上铺")

        val ticket2 = getTicket("上海", "北京")
        ticket2.showTicketInfo("下铺")
    }
}